nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -5622,14 +5625,29 @@ public ProcessorDiagnosticsEntity getProcessorDiagnostics(final String id) {
         nifiMetricsRegistry.setDataPoint(aggregateEvent.getBytesReceived(), "TOTAL_BYTES_RECEIVED",
                 instanceId, "RootProcessGroup", rootPGName, rootPGId, "");
 
+        //Add total task duration for root to the NiFi metrics registry
+        final StatusHistoryEntity rootGPStatusHistory = getProcessGroupStatusHistory(rootPGId);
+        final List<StatusSnapshotDTO> aggregatedStatusHistory = rootGPStatusHistory.getStatusHistory().getAggregateSnapshots();
+        final long taskDuration = aggregatedStatusHistory.isEmpty() ? 0L :
+                aggregatedStatusHistory.get(aggregatedStatusHistory.size() - 1).getStatusMetrics()
```

Comment:
This line is a bit difficult to read because of the inline index calculation based on `size() - 1`.  Is there a reason for always reading the last element as opposed to the first?  Perhaps definition and index variable and including a comment would be helpful.